src/UI/Components/Diagram/shapes.ts

```diff
@@ -0,0 +1,209 @@
+        columns: [],
+        padding: { top: 40, bottom: 10, left: 10, right: 10 },
+        size: { width: 176 },
+        itemMinLabelWidth: 80,
+        itemHeight: 22,
+        itemOffset: 0,
+        itemOverflow: true,
+        attrs: {
+          root: {
+            magnet: false,
+          },
+          body: {
+            stroke: "#FFF",
+            fill: "#FFF",
+            strokeWidth: 1,
+          },
+          header: {
+            fill: "#0066CC",
+            stroke: "#0066CC",
+            strokeWidth: 1,
+          },
+          headerLabel: {
+            fill: "#FFFFFF",
+            fontFamily: "Lekton",
+            textTransform: "uppercase",
+            fontSize: 12,
+            textWrap: {
+              ellipsis: true,
+              height: 30,
+            },
+          },
+          itemBodies_0: {
+            // SVGRect which is an active magnet
+            // Do not use `true` to prevent CSS effects on hover
+            magnet: "item",
+          },
+          group_1: {
+            // let the pointer events propagate to the group_0
+            // which spans over 2 columns
+            pointerEvents: "none",
+          },
+          itemLabels: {
+            fontFamily: "Lekton",
+            fontSize: 10,
+            fill: "#000000",
+            pointerEvents: "none",
+          },
+          itemLabels_1: {
+            fill: "#7F7F7F",
+            fontSize: 10,
+            fontFamily: "Lekton",
+            textAnchor: "end",
+            x: `calc(0.5 * w - 10)`,
+          },
+          itemLabels_keys: {
+            x: `calc(0.5 * w - 30)`,
+          },
+          iconsGroup_1: {
+            // SVGGroup does not accept `x` attribute
+            refX: "50%",
+            refX2: -26,
+          },
+        },
+      },
+      super.defaults
+    );
+  }
+
+  preinitialize(): void {
+    this.markup = [
+      {
+        tagName: "rect",
+        selector: "body",
+      },
+      {
+        tagName: "rect",
+        selector: "header",
+      },
+      {
+        tagName: "rect",
+        selector: "tabColor",
+      },
+      {
+        tagName: "text",
+        selector: "headerLabel",
+      },
+    ];
+  }
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  initialize(...args: any[]) {
+    super.initialize(...args);
+    this._setColumns(this.get("columns"));
+  }
+
+  onColumnsChange() {
+    if (this.hasChanged("columns")) {
+      this._setColumns(this.get("columns"));
+    }
+  }
+
+  setName(name: string, opt?: object) {
+    return this.attr(["headerLabel", "text"], name, opt);
+  }
+
+  getName(): string {
+    return this.attr(["headerLabel", "text"]);
+  }
+  setColumns(data: Array<ColumnData>) {
+    this.set("columns", data);
+    return this;
+  }
+  appendColumns(data: Array<ColumnData>) {
+    this._setColumns(data);
+    return this;
+  }
+
+  toJSON() {
+    const json = super.toJSON();
+    // keeping only the `columns` attribute
+    delete json.items;
```

Comment:
as it's from demo I am not sure what's the complete reasoning for that, I would plan to remove all the unnesseracy methods when we have all base functionalities for operations done, to avoid deleting something that was there in the first place